kds/kds.go

```diff
@@ -667,11 +971,33 @@ func parseTCBURL(u *url.URL) (uint64, error) {
 			setter(uint8(number))
 		}
 	}
-	tcb, err := ComposeTCBParts(parts)
+	tcb, err := t.Compose()
 	if err != nil {
 		return 0, fmt.Errorf("invalid AMD KDS TCB arguments: %v", err)
 	}
-	return uint64(tcb), err
+	return tcb.Raw(), err
+}
+
+func newTCBParts(productLine string) TCBPartsI {
+	switch productLine {
+	case "Milan":
+		return &TCBParts{}
+	case "Genoa":
+		return &TCBParts{}
+	default:
```

Comment:
Probably best to not assume anything beyond Turin and return nil for default if the productLine is not known.